tests/unit/test_workflow_files.py

```diff
@@ -102,25 +103,48 @@ def test_is_valid_run_dir(is_abs_path: bool, tmp_run_dir: Callable):
     assert workflow_files.is_valid_run_dir(Path(prefix, 'foo/bar')) is True
 
 
-def test_check_nested_run_dirs(tmp_run_dir: Callable):
-    """Test that check_nested_run_dirs() raises when a parent dir is a
+def test_check_nested_dirs(tmp_run_dir: Callable):
+    """Test that check_nested_dirs() raises when a parent dir is a
     workflow directory."""
     cylc_run_dir: Path = tmp_run_dir()
     test_dir = cylc_run_dir.joinpath('a/b/c/d/e')
     test_dir.mkdir(parents=True)
     # Parents are not run dirs - ok:
-    check_nested_run_dirs(test_dir)
+    check_nested_dirs(test_dir)
     # Parent contains a run dir but that run dir is not direct ancestor
     # of our test dir - ok:
     tmp_run_dir('a/Z')
-    check_nested_run_dirs(test_dir)
+    check_nested_dirs(test_dir)
     # Now make run dir out of parent - not ok:
     tmp_run_dir('a')
     with pytest.raises(WorkflowFilesError) as exc:
-        check_nested_run_dirs(test_dir)
+        check_nested_dirs(test_dir)
     assert "Nested run directories not allowed" in str(exc.value)
 
 
+@pytest.mark.parametrize(
+    'test_install_path, existing_install_path',
+    [
+        (f'{"child/"*i}', '') for i in range(2, MAX_SCAN_DEPTH + 3)
+    ] + [
+        ('', f'{"child/"*1}')
+    ]
+)
+def test_check_nested_dirs_install_dirs(
+    tmp_path: Path, test_install_path: str, existing_install_path: str
+):
+    """Test that check nested dirs looks both up and down a tree for
+    WorkflowFiles.Install.DIRNAME."""
+    testdir = tmp_path / test_install_path
+    testdir.mkdir(parents=True, exist_ok=True)
+    (tmp_path / existing_install_path / WorkflowFiles.Install.DIRNAME).mkdir(
+        parents=True
+    )
```

Comment:
(For future reference) Note you can supply an `installed` arg to the `tmp_run_dir()` fixture instead of manually `mkdir`ing
https://github.com/cylc/cylc-flow/blob/e69a9a3b372f623ee1bb1ac59eb5582a32ccca0d/tests/unit/conftest.py#L70